src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/BoundaryCondition.hpp

```diff
@@ -0,0 +1,80 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <pup.h>
+
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Domain/BoundaryConditions/None.hpp"
+#include "Domain/BoundaryConditions/Periodic.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Factory.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Factory.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \brief Boundary conditions for the combined Generalized Harmonic and
+/// Valencia GRMHD systems
+namespace grmhd::GhValenciaDivClean::BoundaryConditions {
+/// \cond
+template <typename DerivedGhCondition, typename DerivedValenciaCondition>
+class ProductOfConditions;
+/// \endcond
+
+namespace detail {
+template <typename DerivedGhCondition, typename DerivedValenciaCondition>
+using ProductOfConditionsIfConsistent = tmpl::conditional_t<
+    DerivedGhCondition::bc_type == DerivedValenciaCondition::bc_type,
+    ProductOfConditions<DerivedGhCondition, DerivedValenciaCondition>,
+    tmpl::list<>>;
+
+template <typename GhList, typename ValenciaList>
+struct AllProductConditions;
+
+template <typename GhList, typename... ValenciaConditions>
+struct AllProductConditions<GhList, tmpl::list<ValenciaConditions...>> {
+  using type = tmpl::flatten<tmpl::list<tmpl::transform<
+      GhList, tmpl::bind<ProductOfConditionsIfConsistent, tmpl::_1,
+                         tmpl::pin<ValenciaConditions>>>...>>;
+};
+}  // namespace detail
+
+/// \brief The base class for Generalized Harmonic and Valencia combined
+/// boundary conditions; all boundary conditions for this system must inherit
+/// from this base class.
+class BoundaryCondition : public domain::BoundaryConditions::BoundaryCondition {
+ public:
+  using creatable_classes = tmpl::push_back<
```

Comment:
could you add  a short comment like `remove the periodic and none BCs from the creatable classes of the individual systems`?